### ISSUE TITLE
Validator runtime error when trying to define an inverse functional object property

### ISSUE THREAD

**Issue by KonradHoeffner** [RDFLib/pySHACL#114]
```
  File "/home/konrad/.local/lib/python3.10/site-packages/pyshacl/cli.py", line 196, in main
    is_conform, v_graph, v_text = validate(args.data, **validator_kwargs)
  File "/home/konrad/.local/lib/python3.10/site-packages/pyshacl/validate.py", line 419, in validate
    conforms, report_graph, report_text = validator.run()
  File "/home/konrad/.local/lib/python3.10/site-packages/pyshacl/validate.py", line 222, in run
    shapes = self.shacl_graph.shapes  # This property getter triggers shapes harvest.
  File "/home/konrad/.local/lib/python3.10/site-packages/pyshacl/shapes_graph.py", line 164, in shapes
    self._build_node_shape_cache()
  File "/home/konrad/.local/lib/python3.10/site-packages/pyshacl/shapes_graph.py", line 191, in _build_node_shape_cache
    raise ShapeLoadError(


Validator encountered a Runtime Error. Please report this to the PySHACL issue tracker.
```

This error occurs when adding the path to the following shape:

```turtle
:RepositoryRangeShape a sh:NodeShape;
	sh:targetObjectsOf :repository;
	sh:path [
			sh:path [ sh:inversePath :repository ];
			sh:maxCount 1
	];
	sh:nodeKind sh:IRI.
```

The goal behind the shape is the translation of `:repository rdfs:range rdfs:Resource` (any URI) to SHACL with the additional cardinality constraint that each value may only be used by at most one software product (domain omitted here).
I assume that the problem occurs because the value of sh:targetObjectsOf is the same as the object of the sh:inversePath triple.
The problem occurs even when I remove the `sh:nodeKind` triple.
A similar construct is given in the accepted answer https://stackoverflow.com/a/63365996/398963.

**Comment by ashleysommer** [comment]
Hi @KonradHoeffner 
Looks like its a typo in your NodeShape definition. It should be:
```bash
:RepositoryRangeShape a sh:NodeShape;
	sh:targetObjectsOf :repository;
	sh:property [
		sh:path [ sh:inversePath :repository ];
		sh:maxCount 1
	] .
```

**Comment by KonradHoeffner** [comment]
@ashleysommer: Thanks, that indeed fixed the problem!
However, is it really intended that pySHACL throws a Runtime Error here? As a user I would expect a Runtime Error to indicate that there is something wrong with pySHACL itself, not my SHACL file.

**Comment by ashleysommer** [comment]
You're right, it throws a `ShapeLoadError()` at that point, which should give you a good explanation of what the problem is and where to look for a solution. I'm not sure why it crashes the application instead of giving you the correct error message.

**Comment by KonradHoeffner** [comment]
The error seems to occur at the following place in the code in `shapes_graph.py`

```py
   def _build_node_shape_cache(self):
        """             
        :returns: None  
        :rtype: NoneType
        """
        g = self.graph  
        defined_node_shapes = set(g.subjects(RDF_type, SH_NodeShape))
        for s in defined_node_shapes:
            path_vals = list(g.objects(s, SH_path))
            if len(path_vals) > 0:
                # TODO:coverage: we don't have any tests for invalid shapes
                raise ShapeLoadError(
                    "A shape defined as a NodeShape cannot be the subject of a 'sh:path' predicate.", # line 191, the one that causes the error
                    "https://www.w3.org/TR/shacl/#node-shapes",
                )    
```
You can use the shape that caused it as a test case if you want, repeated here:

```py
:RepositoryRangeShape a sh:NodeShape;
	sh:targetObjectsOf :repository;
	sh:path [
			sh:path [ sh:inversePath :repository ];
			sh:maxCount 1
	];
	sh:nodeKind sh:IRI.
```

**Comment by ashleysommer** [comment]
@KonradHoeffner 
The issue with error reporting on a ShapeLoadError (as described above) is now fixed as of PySHACL v0.19.0.
I'll go ahead and close this issue now.
